File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.py
def solution(s, skip, index):
    answer = ''
    alp = "abcdefghijklmnopqrstuvwxyz"
    
    for ch in skip: # ch => skip의 문자 하나하나
        if ch in alp:
            alp = alp.replace(ch, "") # 알파벳 안에 skip 문자들 제거
    
    for word in s:
        tmp = alp[(alp.index(word) + index) % len(alp)] # 각문자의 시작위치 + index를 alpha의 길이로 나눈 나머지를 알파벳으로 변환
        answer += tmp
    return answer

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.java
import java.util.*;

class Solution {
    public String solution(String s, String skip, int index) {
        String answer = "";
        String alpha = "abcdefghijklmnopqrstuvwxyz";
        
        Map<String, Integer> map = new HashMap();
        
        for (String sk : skip.split("")){
            map.put(sk, 1);
        }
        
        for (String c : s.split("")){
            int count = index;
            int idx = 0;
            
            for (int i = 0; i < alpha.length(); i++){
                if (c.equals(String.valueOf(alpha.charAt(i)))){
                    idx = i;
                    break;
                }
            }

            while (count > 0) {
                idx++;
                count--;
                
                if(idx >= alpha.length())
                    idx = 0;
                
                if (map.containsKey(String.valueOf(alpha.charAt(idx)))){
                    count++;
                    continue;
                }   
            }
            
            if (idx == alpha.length())
                idx = 0;
            
            answer+= alpha.charAt(idx);
        }
        return answer;
    }
}